### ISSUE TITLE
Getting Nan's for the Loss function

### ISSUE THREAD

**Issue by SammieH21** [BuffetCodes/Linear-Regression-using-PyTorch-CPP#11]
Hello! When I run the code for both a custom data set and the default data set, I am getting "nan's" returned for the loss function.

I ran make -f Makefile and subsequently ./bin/lr-example,

Here is a snippet of the result:

Loss: 644.398
Loss: 1.49074e+07
Loss: 1.02868e+38
Loss: -nan
Loss: -nan
Loss: -nan
Loss: -nan

It appears to be the case that the model's loss got worse after training instead of better over itme

**Comment by krshrimali** [comment]
Hi, @SammieH21 - Can you please share which PyTorch version are you using? Sorry for the late response btw.
